mlrun/datastore/datastore.py

```diff
@@ -185,7 +189,14 @@ def object(
         store, subpath = self.get_or_create_store(
             url, secrets=secrets, project_name=project
         )
-        return DataItem(key, store, subpath, url, meta=meta, artifact_url=artifact_url)
+        return DataItem(
+            key,
+            store,
+            subpath,
+            store.url + subpath,
```

Comment:
Are you sure all the stores has been tested with this change? I would use 'store.url + subpath if url and url.startswith("ds://") else url' - if the purpose is it to handle datastore profiles